analyzers/tests/SonarAnalyzer.Test/TestCases/NamedPlaceholdersShouldBeUnique.cs

```diff
@@ -0,0 +1,300 @@
+﻿using System;
+
+namespace MicrosoftTests
+{
+    using Microsoft.Extensions.Logging;
+
+    public class Program
+    {
+        void Compliant(ILogger logger, string foo, string bar)
+        {
+            var args = new object[] { foo, bar };
+
+            Console.WriteLine("Hey {foo} and {foo}"); // Compliant
+
+            // Do not raise on duplicate indexes
+            logger.LogInformation("Hey {0} and {0}", foo, bar);             // Compliant
+            logger.LogDebug("Hey {0} and {foo} and {0}", foo, foo, bar);    // Compliant
+            logger.LogTrace("Hey {0} and {0}", foo, bar);                   // Compliant
+            logger.LogCritical("Hey {0} and {0}", foo, bar);                // Compliant
+            logger.LogWarning("Hey {0} and {0}", foo, bar);                 // Compliant
+            logger.LogError("Hey {0} and {0}", foo, bar);                   // Compliant
```

Comment:
We do have plenty of repetition in the logs. Could you please extract parameterized tests as we have in the case of https://github.com/SonarSource/sonar-dotnet/blob/56a6076a83dc7fe3e0cbbf22f197f36588ed3091/analyzers/tests/SonarAnalyzer.Test/Rules/ExceptionsShouldBeLoggedTest.cs and https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/tests/SonarAnalyzer.Test/Rules/UseConstantLoggingTemplateTest.cs?